pyproject.toml

```diff
@@ -0,0 +1,66 @@
+[build-system]
```

Comment:
This triggers a big "tech debt/duplication/combinatorial explosion/confusion" red flag for me. This duplicates ./flake8 and test/static-code, and introduces all kinds of new dependencies, like `tox` and whatnot. Aside from the fact that they need to be documented and added to our unit-tests container, I must say I dislike wrappers like `tox` -- it is often rather hard to peel them off to debug a single test case, and invoke the plain test .py file directly. It also isn't integrated into `make check`, and instead starts to build a parallel toolset. I don't see anything in test_transport.py which couldn't/shouldn't work with e.g. `PYTHONPATH=src python3 -m unittest discover`?

Also, this isn't really a Python *module* that we'd ever want to publish or support -- the fact that the bridge is written in Python ought to be an implementation detail. Do you introduce this file only for putting in the parameters for static code checkers? That I can agree with, but then we don't need most of the other metadata and setuptools config here.

Reply:
This mostly exists as the one central location for config settings for:
 - mypy
 - pylint
 - pytest
 - coverage
 - tox
 - probably something I forgot

but also, the project metadata is needed for building the package, which is required for tox to construct the virtual environment for testing.  I agree that we're probably not going to want to ship this package as such (ie: on PyPI) any time soon.  That's why, for example, the version is hardcoded to 0.

Reply:
Note also: assuming you have `pytest` installed, you can run `pytest` directly, and that will work, out of the source tree.  And indeed: that's because of the configuration in `pyproject.toml`.

Reply:
One more note: I'd like to be able to do the python testing without having configured the project.  In particular, I find the way that we currently import `systemd_ctypes` (which I've intentionally avoided testing, until now) to be a bit of a hack.  Being able to specify that, python-style in a `requirements.txt`, for example, seems a lot nicer to me, and it's precisely I'm planning to do as a next step.

We could, of course, try to figure out a way to launch `pytest` via our existing build system, but ... eh...

I think our 1:1 today will be interesting :)

Reply:
> This duplicates ./flake8 and test/static-code

A more careful inspection: It doesn't actually; I mis-read e.g. the `max-line-length`, but that's for pylint. It'd be nice to move .flake8 into pyproject.toml, but that's another story, difficult, and not a blocker.

Reply:
vulture has pyproject support https://github.com/jendrikseipp/vulture#configuration